content/developer/reference/user_interface/views/include/form.rst

```diff
@@ -0,0 +1,778 @@
+
+:disable_autofocus:
+  boolean_
+
+  Disable automatic focus of the first field in the view.
+
+.. _reference/user_interface/views/form/semantic:
+
+Semantic components
+-------------------
+
+Semantic components tie into and allow interaction with the Odoo system.
+Placeholders are denoted in all caps.
+
+.. _reference/user_interface/views/form/field:
+
+<field>: render formatted values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: xml
+
+  <form>
+    <field name="FIELD_NAME"/>
+  </form>
+
+renders (and allow editing of, possibly) a single field of the current
+record. Using several times a field in a form view is supported and the fields
+can receive different values for modifiers 'invisible' and 'readonly'. However,
+the behavior is not guaranteed when several fields exist with different values
+for modifier 'required'. ``<field>`` can have the following attributes:
+
+:name:
+  string_ :ref:`model <reference/orm/model>` field name (mandatory)
+
+  the name of the field to render
+
+:id:
+  string_
+
+  the node id. Useful when there are several occurrences of the same field in
+  the view (see ``label`` component below). Default is the field name.
+
+:widget:
+  string_
+
+  fields have a default rendering based on their type
+  (e.g. :class:`~odoo.fields.Char`, :class:`~odoo.fields.Many2one`).
+
+  The ``widget`` attributes allows using a different rendering method and context.
+  See more information in :ref:`reference/js/widgets`
+
+  .. code-block:: xml
+
+    <field name="tag_ids" widget="many2many_tags"/>
+
+:options:
+  :ref:`python expression <user_interface/views/python_expression>` that defines a dict_
+
+  JSON object specifying configuration option for the field's widget
+  (including default widgets)
+
+  .. code-block:: xml
+
+    <field name="tag_ids" widget="many2many_tags" options="{'color_field': 'FIELD_NAME', 'no_quick_create': True}"/>
+
+:groups:
+  `Comma-separated values`_ whose choices are the :class:`~odoo.addons.base.models.res_users.Groups` reference
+
+  only displays the field for specific users
+
+  .. code-block:: xml
+
+    <field name="fname" groups="base.group_no_one,!base.group_multi_company"/>
+
+:domain:
+  :ref:`python expression <user_interface/views/python_expression>` that defines a :ref:`reference/orm/domains`
+
+  for relational fields only, filters to apply when displaying existing
+  records for selection
+
+  .. code-block:: xml
+
+    <field name="fname_id" domain="[('fname_a', '=', parent.fname_b)]"/>
+
+:context:
+  :ref:`python expression <user_interface/views/python_expression>` [#python_expression] that defines a dict_
+
+  for relational fields only, context to pass when fetching possible values
+
+  .. code-block:: xml
+
+    <field name="fname" context="{
+        'TYPE_view_ref': 'ADDON.MODEL_view_TYPE',
+        'group_by': 'FIELD_NAME',
+        'default_FIELD_NAME': ANY,
+        'search_default_FIELD_NAME': True,
+        'OTHER_BUSINESS_KEY': ANY,
+      }"/>
```

Comment:
is there some place where we explain the *special* behavior of certain context keys, like `default_* ?
`